src/util.cpp

```diff
@@ -61,7 +61,14 @@ std::string intToString(int i) {
 
 std::string urlDecode(std::string input) {
   std::size_t idxReplaced = 0;
-  std::size_t idxFound = input.find('%');
+  // First replace + by space
+  std::size_t idxFound = input.find('+');
+  while (idxFound != std::string::npos) {
+    input.replace(idxFound, 1, 1, ' ');
+    idxFound = input.find('+', idxFound);
```

Comment:
```suggestion
    idxFound = input.find('+', idxFound + 1);
```

Just to not look at the already processed chars again.